src/chrdev.rs

```diff
@@ -142,8 +142,13 @@ unsafe extern "C" fn read_callback<T: FileOperations>(
         Err(e) => return e.to_kernel_errno() as c_types::c_ssize_t,
     };
     let f = &*((*file).private_data as *const T);
-    // TODO: Pass offset to read()?
-    match f.read(&mut data) {
+    // No FMODE_UNSIGNED_OFFSET support, so offset must be in [0, 2^63).
+    // See discussion in #113
+    let positive_offset = match (*offset).try_into() {
+        Ok(v) => v,
+        Err(_) => return Error::EINVAL.to_kernel_errno() as c_types:c_ssize_t,
```

Comment:
`try_into().unwrap()`? Dunno if that's any better.

Reply:
Hm, looks like it compiles out, sure.

We should probably have a style guide about how we feel about `as` vs. `unwrap()`. I think my gut feeling is, try not to have the problem, but if forced to pick, `unwrap()` is at least safe.